spec/shared/controllers/hiding_columns_tenant.rb

```diff
@@ -0,0 +1,39 @@
+shared_examples 'hiding tenant columns' do
+
+  before do
+    allow(controller).to receive(:render)
+    login_as user
+  end
+
+  subject do
+    expect(controller).to receive(:get_db_view).and_return(report)
+    controller.send(:report_data)
+    controller.send(:view_to_hash, assigns(:view))
+  end
+
+  context "when not admin" do
+    let(:user) { FactoryBot.create(:user) }
```

Comment:
If you move up `user` a level above like this:
```
let(:user) { FactoryBot.create(user_type) }
```
Then you can just define `user_type` in both examples. It doesn't seem that nice, but you'll understand in the 3rd step :wink: